File: backend/api/portals.go
package api

import (
	"github.com/gofiber/fiber/v2"
	"os"
	"sort"
	"strings"
)

func Portals(c *fiber.Ctx) error {

	var list []string

	// get dir list
	dirs, err := os.ReadDir("./portal")
	if err != nil {
		return err
	}

	for _, dir := range dirs {
		if !dir.IsDir() {
			continue
		}

		list = append(list, dir.Name())
	}

	// alphabetically sort list
	sort.Slice(list, func(i, j int) bool {
		return strings.ToLower(list[i]) < strings.ToLower(list[j])
	})

	return c.Status(200).JSON(Response{Message: "success", Data: list})
}

File: frontend/src/lib/portals.ts
import type {response} from "./var.ts"
import {api} from "./var.ts"

export const getPortals = async (): Promise<string[]> => {
	const rsp: Response = await fetch(api + "api/portals", {
		credentials: "include"
	})

	const rspJson: response = await rsp.json()

	if (rsp.status === 200 && rspJson.data) {
		return rspJson.data as string[]
	}

	return []
}